net/tstun/wrap.go

```diff
@@ -481,6 +494,16 @@ func (t *Wrapper) filterIn(buf []byte) filter.Response {
 				f(data)
 			}
 		}
+	case ipproto.UDP:
+		if pkt := p.Payload(); disco.LooksLikeDiscoWrapper(pkt) {
+			discoFrom := pkt[len(disco.Magic):][:len(tailcfg.DiscoKey{})]
```

Comment:
Can we add a package disco method to extract `discoFrom`, rather than having tstun know about the structure of disco packets?

Reply:
done